### ISSUE TITLE
📝 [md] Continuation line not detected

### ISSUE THREAD

**Issue by ematipico** [biomejs/biome#10347]
### Environment information

main

### Configuration

```JSON

```

### Playground link

https://biomejs.dev/playground/?lineWidth=120&tab=syntax&pane=Diagnostics&code=MQAuACAAIABmAG8AbwAKAAoAIAAgACAAIAAtACAAYgBhAHIACgAgACAACgAKADEALgAgAGYAbwBvAAoAIAAgACAALQAgAGIAYQByAAoA&language=md

### Expected result

The second example doesn't have a continuous indentation, even though it should. This should match what we have in the grammar and in common mark. 

Instead, it's parsed as pre marker indent, which I think it's a bug. 

cc @jfmcdowell 

FYI this is another bug that blocked me. 

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by jfmcdowell** [comment]
@ematipico I am working on a fix
